helix-view/src/editor.rs

```diff
@@ -1419,7 +1421,22 @@ impl Editor {
         // via stream.then() ? then push into main future
 
         let path = path.map(|path| path.into());
+        let config = self.config();
         let doc = doc_mut!(self, &doc_id);
+
+        // handle adding newline at the end of the file if there is none
+        let newline = doc.line_ending.as_str();
+        let text = doc.text();
+        let doc_len = text.len_chars();
+        if config.newline_at_eof && get_line_ending(&text.slice(..)).is_none() {
+            let view = view_mut!(self);
```

Comment:
This will fetch the currently focused view which may not be the view that's looking at the document. This can cause a panic with this feature enabled: open a split with two different files without newline endings and `:wa`.

I think it might make sense to solve this in `helix-term::commands::typed` in `write_impl` and `write_all_impl` rather than here in Editor. We probably also do not want to add a trailing newline when documents are being auto-formatted because we may end up fighting with the auto-formatter (language server or external command)